cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -96,18 +103,180 @@ def set_default_table(cls, values):
         return values
 
     @root_validator  # not really a validator, more like a post-processor
+    @classmethod
     def fixup_lookup_names(cls, v):
         "Allow lookup objects to know the key they were attached to in the mapping file."
         for name, lookup in v["lookups"].items():
             lookup.name = name
         return v
 
+    @staticmethod
+    def _is_injectable(element: str) -> bool:
+        return element.count("__") == 1
+
+    def _get_permission_type(self, operation: DataOperationType) -> str:
+        if operation is DataOperationType.QUERY:
+            return "queryable"
+        if operation is DataOperationType.INSERT and self.action == "update":
+            return "updateable"
+
+        return "createable"
+
+    def _check_object_permission(
+        self, global_describe: Dict, sobject: str, operation: DataOperationType
+    ):
+        perm = self._get_permission_type(operation)
+        return sobject in global_describe and global_describe[sobject][perm]
+
+    def _check_field_permission(
+        self, describe: Dict, field: str, operation: DataOperationType
+    ):
+        perm = self._get_permission_type(operation)
+        # Fields don't have "queryable" permission.
+        return field in describe and (
+            describe[field].get(perm) if perm in describe[field] else True
+        )
```

Comment:
No need to do this now, but at some point it might make sense to have our own DescribeSObjectResult class which adapts the JSON result to provide helper methods for checking things like this.